packages/debug/src/browser/debug-session.tsx

```diff
@@ -512,13 +512,36 @@ export class DebugSession implements CompositeTreeElement {
                 new DebugBreakpoint(data, this.labelProvider, this.breakpoints, this.editorManager, this)
             );
             const enabled = all.filter(b => b.enabled);
-            const response = await this.sendRequest('setBreakpoints', {
-                source: source.raw,
-                sourceModified,
-                breakpoints: enabled.map(({ origin }) => origin.raw)
-            });
-            response.body.breakpoints.map((raw, index) => enabled[index].update({ raw }));
-            this.setBreakpoints(affectedUri, all);
+
+            try {
+                const response = await this.sendRequest('setBreakpoints', {
+                    source: source.raw,
+                    sourceModified,
+                    breakpoints: enabled.map(({ origin }) => origin.raw)
+                });
+                response.body.breakpoints.map((raw, index) => enabled[index].update({ raw }));
+            } catch (error) {
+                // could be error or promise rejection of DebugProtocol.SetBreakpointsResponse
+                if (error instanceof Error) {
+                    console.error(`Error setting breakpoints: ${error.message}`);
```

Comment:
`console.error('Error setting breakpoints:', error);` would be better

`error.message` does not contain stack trace information